modules/rostests/apitests/shell32/CMyComputer.cpp

```diff
@@ -92,13 +94,16 @@ START_TEST(CMyComputer)
         return;
     }
 
-    /* second create should give us a pointer to the same object */
-    hr = CoCreateInstance(CLSID_MyComputer,
-                          NULL,
-                          CLSCTX_INPROC_SERVER,
-                          IID_PPV_ARG(IShellFolder2, &psf2_2));
-    ok(hr == S_OK, "hr = %lx\n", hr);
-    ok(psf2 == psf2_2, "Expected %p == %p\n", static_cast<PVOID>(psf2), static_cast<PVOID>(psf2_2));
+    if (GetNTVersion() < _WIN32_WINNT_VISTA)
+    {
+        /* NT5.x returns a pointer to the same object but not NT6.x and newer. */
+        hr = CoCreateInstance(CLSID_MyComputer,
+                              NULL,
+                              CLSCTX_INPROC_SERVER,
+                              IID_PPV_ARG(IShellFolder2, &psf2_2));
+        ok(hr == S_OK, "hr = %lx\n", hr);
+        ok(psf2 == psf2_2, "Expected %p == %p\n", static_cast<PVOID>(psf2), static_cast<PVOID>(psf2_2));
+    }
```

Comment:
But then, the TestShellFolder/TestInitialize calls done below are using uninitialized pointers in Vista+?